actix-server/src/worker.rs

```diff
@@ -159,26 +159,61 @@ enum WorkerServiceStatus {
     Stopped,
 }
 
+// config for worker behavior passed from server builder.
+#[derive(Copy, Clone)]
+pub(crate) struct ServerWorkerConfig {
+    shutdown_timeout: Duration,
+    max_blocking_threads: usize,
+}
+
+impl Default for ServerWorkerConfig {
+    fn default() -> Self {
+        Self {
+            shutdown_timeout: Duration::from_secs(30),
+            // default max_blocking_threads is the same as tokio's default
+            max_blocking_threads: 512,
```

Comment:
I think it should be a bit lower to account for common case of multi-worker so as to not potentially blow up thread count. 512 / 4 sounds good. So 128.

Reply:
What about divide 512 with the core count num_cpus returns.

Reply:
good idea, i guess we have that as a dep anyway